### PR TITLE
Add real dark gray from the 256 color palette

### LINKED ISSUE #57385: Do not expect bold black to be drawn as dark gray
### Describe the bug

A user just asked for help in the IRC channel, stating they could not see the following colors because they are solid black:  c_black (normal); c_dark_gray (normal); i_black (inverted); i_dark_gray (inverted)

Ignore the black since it is supposed to be invisible, because dark_gray is the actual issue here. https://github.com/kovidgoyal/kitty/issues/197 demonstrates this quite well. Alacritty is affected, too but there is the `draw_bold_text_with_bright_colors: true` option, luckily.

image

This is a screenshot from kitty, where there is no option to control the behavior. Note the invisible roads, parking lots and weird rivers.

### Steps to reproduce

1. Start the game in either kitty or alacritty.
2. Check your map or any other UI element using dark gray.
3. Go quaff from a fountain or put on a ring of see invisible to be able to see those roads.

### Expected behavior

Dark gray is not bold black. Light blue is not bold blue. There are proper color codes luckily.

Relevant: #880

### Screenshots

_No response_

### Versions and configuration

- experimental

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SJonesy**
#### Summary
Bugfixes "Fix dark gray in the ncurses client for terminal emulators that support 256 colors."

#### Purpose of change
Fixes #57385

#### Describe the solution
The ncurses client now uses number 237 on the 256 color palette (if available in the terminal emulator) for dark gray instead of using black with the intensity bit set. This makes it so many terminal emulators will no longer require using the "interpret bold colors as bright" setting globally. The ncurses client does this by detecting if the terminal emulator being used supports >= 256 colors and this is exposed in the catacurses namespace.

#### Describe alternatives you've considered
The dark gray color could realistically be anywhere in the range of like 235-239. I like 237 the best.

#### Testing
Opened the client and checked both Settings->Colors and a road in game to make sure the dark gray displayed correctly. I also verified a few other places like inside a dark house and the crafting menu.

#### Additional context
We have a color pair budget of 256 from ncurses, and this change adds the 88th.

### CODE REVIEW

**Comment by Zireael07** [comment]
Plz edit so that there is no fictious link to issue 237 (e.g. by putting a space between hash and number)

**Comment by SJonesy** [comment]
> Plz edit so that there is no fictious link to issue 237 (e.g. by putting a space between hash and number)

Done!

File: src/ncurses_def.cpp [hunk 1]
```diff
@@ -265,6 +265,11 @@ void catacurses::init_interface()
     init_colors();
 }
 
+bool catacurses::supports_256_colors()
+{
+    return( COLORS >= 256 );
```
Comment (ZhilkinSerg) [inline src/ncurses_def.cpp:270]:
```suggestion
    return COLORS >= 256;
```

File: src/wincurse.cpp [hunk 2]
```diff
@@ -656,6 +656,11 @@ void catacurses::init_interface()
     initialized = true;
 }
 
+bool catacurses::supports_256_colors()
+{
+    return( COLORS >= 256 );
```
Comment (ZhilkinSerg) [inline src/wincurse.cpp:661]:
```suggestion
    return COLORS >= 256;
```

**Pull request merged**